### PR TITLE
(FACT-2697) Add Xen fact

### PR DESCRIPTION
**Pull request by IrimieBogdan**
Added `Xen` fact for linux.

### CODE REVIEW

File: spec/facter/facts/linux/hypervisors/xen_spec.rb [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+          expect(Facter::Resolvers::Lspci).to have_received(:resolve).with(:vm).twice
+        end
+
+        it 'returns xen' do
+          expect(fact.call_the_resolver).to be_an_instance_of(Facter::ResolvedFact)
+            .and have_attributes(name: 'hypervisors.xen', value: { 'context' => 'hvm', 'privileged' => true })
+        end
+      end
+
+      context 'when hvm context' do
```
Comment (oanatmaria) [inline spec/facter/facts/linux/hypervisors/xen_spec.rb:88]: this is duplicated with 'when VirtWhat resolver returns xen'

File: spec/facter/facts/linux/hypervisors/xen_spec.rb [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+          allow(Facter::Resolvers::Xen).to receive(:resolve).with(:privileged).and_return(true)
+        end
+
+        it 'returns privileged xen' do
+          expect(fact.call_the_resolver).to be_an_instance_of(Facter::ResolvedFact)
+            .and have_attributes(name: 'hypervisors.xen', value: { 'context' => 'hvm', 'privileged' => true })
+        end
+      end
+
+      context 'when unprivileged' do
```
Comment (oanatmaria) [inline spec/facter/facts/linux/hypervisors/xen_spec.rb:128]: this is duplicated with 'when VirtWhat resolver returns xen'

File: spec/facter/facts/linux/hypervisors/xen_spec.rb (hunk 1 above)
Comment (IrimieBogdan) [reply spec/facter/facts/linux/hypervisors/xen_spec.rb]: removed duplicated code

File: spec/facter/facts/linux/hypervisors/xen_spec.rb (hunk 2 above)
Comment (IrimieBogdan) [reply spec/facter/facts/linux/hypervisors/xen_spec.rb]: removed duplicated code

**Pull request merged**